h/auth/services.py

```diff
@@ -61,26 +65,26 @@ def verify_ticket(self, principal, ticket_id):
         if ticket_id is None:
             return False
 
-        # Since we load and then update the AuthTicket.expires on every request,
-        # we optimise it with a `UPDATE ... RETURNING` query, doing both of
-        # these in one. When the returned `userid` is `None` then the
-        # authentication failed, otherwise it succeeded. Doing a query like this
-        # will not mark the session as dirty, and thus the transaction will
-        # get rolled back at the end, unless we found a userid, at which point
-        # we manually mark the session as changed.
-        ticket_query = (sa.update(AuthTicket.__table__)
-                        .where(sa.and_(AuthTicket.id == ticket_id,
-                                       AuthTicket.user_userid == principal,
-                                       AuthTicket.expires > sa.func.now()))
-                        .values(expires=(sa.func.now() + TICKET_TTL))
-                        .returning(AuthTicket.user_userid))
-        self._userid = self.session.execute(ticket_query).scalar()
-
-        if self._userid:
-            mark_changed(self.session)
-            return True
-
-        return False
+        ticket = self.session.query(AuthTicket) \
+            .filter(AuthTicket.id == ticket_id,
+                    AuthTicket.user_userid == principal,
+                    AuthTicket.expires > sa.func.now()) \
+            .first()
+
+        if ticket is None:
+            return False
+
+        self._userid = ticket.user_userid
+
+        # We don't want to update the `expires` column of an auth ticket on
+        # every single request. So here we check that the difference would be
+        # bigger than configured in the `NEW_EXPIRY_THRESHOLD` constant, and
+        # only then update the auth ticket.
+        new_expires = datetime.datetime.utcnow() + TICKET_TTL
+        if (new_expires - ticket.expires) > NEW_EXPIRY_THRESHOLD:
```

Comment:
Sure, I was thinking of doing it like that, but decided against it in case the `updated` column gets updated due to another column getting changed?